prusti-viper/src/encoder/counterexample.rs

```
@@ -122,6 +123,67 @@ impl Entry {
+                        };
+                } else {
+                    //impossible to reach
+                    Entry::Unknown
+                }
+            },
+            (Entry::Enum { super_name: super_name1, name: name1,  field_entries: field_entries1 },
+                Entry::Enum { super_name: super_name2, name: name2,  field_entries: field_entries2 }) => {
+                    if super_name1.to_string() == super_name2.to_string() {
+                        if name1.to_string() == "?".to_string() {
+                            return other.clone();
+                        }
+
+                        if name1.to_string() == name2.to_string() && field_entries1.len() == field_entries2.len() {
+                            let mut other_iter = field_entries2.iter();
+                            let new_field_entries = field_entries1.iter().map(|x| {
+                                let next = other_iter.next().unwrap();
+                                (x.0.clone(), x.1.merge(&next.1))
+                            }).collect();
+                            return Entry::Enum{
+                                super_name: super_name1.to_string(),
+                                name: name1.to_string(),
+                                field_entries: new_field_entries,
+                            };
+                        }
+                    } 
+                    self.clone()
+                },
+            (Entry::Tuple(entries1), Entry::Tuple(entries2)) => {
+                if entries1.len() == entries2.len(){
+                    let mut other_iter = entries2.iter();
+                        let new_entries = entries1.iter().map(|x| {
+                            let next = other_iter.next().unwrap();
+                            x.merge(next)
+                        }).collect();
+                        Entry::Tuple(new_entries)
+                } else {
+                    self.clone()
```

Why this silent failure?